src/turbomind/models/llama/LlamaBatch.cc

```diff
@@ -627,11 +627,11 @@ void LlamaBatch<T>::Initialize(GenerationState& g)
     g.partial_context_legnth = partial_len;
     g.unique_ids             = std::move(unique_ids);
     g.finished_count         = 0;
+    g.skip_init_sampling     = skip_init_sampling;
 
     if (!skip_init_sampling) {
         g.max_init_ctx_len = max_context_len;
         g.step             = max_context_len;
-        InitializeSampling(g);
```

Comment:
Is there any necessity that we move `intializeSampling` from `Initialize` to `Forward`?